src/test/java/org/apache/brooklyn/cloudfoundry/entity/VanillaCloudFoundryApplicationLiveTest.java

```diff
@@ -58,7 +63,65 @@ public void testDeployApplicationWithoutDomain() throws IOException {
+
+        final VanillaCloudfoundryApplication entity =
+                app.createAndManageChild(EntitySpec.create(VanillaCloudfoundryApplication.class)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_NAME, applicationName)
+                        .configure(VanillaCloudfoundryApplication.ARTIFACT_PATH, APPLICATION_ARTIFACT_URL)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_DOMAIN, DEFAULT_DOMAIN)
+                        .configure(VanillaCloudfoundryApplication.ENV, env)
+                        .configure(VanillaCloudfoundryApplication.BUILDPACK, JAVA_BUILDPACK));
+
+        startAndCheckEntitySensors(entity, cloudFoundryPaasLocation);
+        assertEquals(entity.getAttribute(VanillaCloudfoundryApplication.APPLICATION_ENV), env);
+    }
+
+    @Test(groups = {"Live"})
+    @SuppressWarnings("unchecked")
+    public void testSetEnvEffector() throws IOException {
+        final VanillaCloudfoundryApplication entity =
+                app.createAndManageChild(EntitySpec.create(VanillaCloudfoundryApplication.class)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_NAME, applicationName)
+                        .configure(VanillaCloudfoundryApplication.ARTIFACT_PATH, APPLICATION_ARTIFACT_URL)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_DOMAIN, DEFAULT_DOMAIN)
+                        .configure(VanillaCloudfoundryApplication.BUILDPACK, JAVA_BUILDPACK));
+
+        startAndCheckEntitySensors(entity, cloudFoundryPaasLocation);
+        assertTrue(entity.getAttribute(VanillaCloudfoundryApplication.APPLICATION_ENV).isEmpty());
+        entity.setEnv("k1", "v1");
+        assertFalse(entity.getAttribute(VanillaCloudfoundryApplication.APPLICATION_ENV).isEmpty());
+        Map<String, String> envs =
+                entity.getAttribute(VanillaCloudfoundryApplication.APPLICATION_ENV);
+        assertTrue(envs.containsKey("k1"));
+        assertEquals(envs.get("k1"), "v1");
+    }
+
+    @Test(groups = {"Live"})
+    public void testModifyApplicationResources() throws IOException {
+        final VanillaCloudfoundryApplication entity =
+                app.createAndManageChild(EntitySpec.create(VanillaCloudfoundryApplication.class)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_NAME, applicationName)
+                        .configure(VanillaCloudfoundryApplication.ARTIFACT_PATH, APPLICATION_ARTIFACT_URL)
+                        .configure(VanillaCloudfoundryApplication.APPLICATION_DOMAIN, DEFAULT_DOMAIN)
+                        .configure(VanillaCloudfoundryApplication.BUILDPACK, JAVA_BUILDPACK));
+
+        startAndCheckEntitySensorsAndDefaultProfile(entity, cloudFoundryPaasLocation);
+
+        entity.setMemory(DOUBLE_MEMORY);
+        assertEquals(entity.getAttribute(VanillaCloudfoundryApplication.ALLOCATED_MEMORY).intValue(),
+                DOUBLE_MEMORY);
```

Comment:
Thanks a lot @andreaturli. 
In any case, I would like to refactor the test in order to avoid duplications :)